File: Assignment no. 55/Q2.c
//Reverse the rows of the matrix

#include<stdio.h>
#include<stdlib.h>
#include<malloc.h>

void ReverseRow(int **Arr,int iRow,int iCol)
{
	int i=0,j=0,l=0,temp;
	for(i=0;i<iRow;i++)
	{
		for(j=(iCol-1),l=0;l<j;j--,l++)
		{
            temp=Arr[i][l];
		    Arr[i][l]=Arr[i][j];
		    Arr[i][j]=temp;
		}

	}

}

int main()
{
	int **p=NULL,iRow=0,iCol=0,i=0,j=0;

	printf("Enter the number of rows:\n");
	scanf("%d",&iRow);

	printf("Enter the number of columns:\n");
	scanf("%d",&iCol);

	p=(int **)malloc(iRow*sizeof(int *));

	for(i=0;i<iRow;i++)
	{
		p[i]=(int *)malloc(iCol*sizeof(int));
	}

	printf("Enter the elements:\n");
	for(i=0;i<iRow;i++)
	{
		for(j=0;j<iCol;j++)
		{
			scanf("%d",&p[i][j]);
		}
	}

	printf("Entered elements are:\n");
	for(i=0;i<iRow;i++)
	{
		for(j=0;j<iCol;j++)
		{
			printf("%d\t",p[i][j]);
		}
		printf("\n");
	}

    ReverseRow(p,iRow,iCol);

    printf("\nAfter swapping :\n");
	for(i=0;i<iRow;i++)
	{
		for(j=0;j<iCol;j++)
		{
			printf("%d\t",p[i][j]);
		}
		printf("\n");
	}

	for(i=0;i<iRow;i++)
    {
        free(p[i]);
    }

    free(p);

	return 0;
}

File: Assignment no. 70/Q2.java
import java.util.Scanner;

class Q2
{
	public static void main(String arg[])
	{
		Scanner sobj=new Scanner(System.in);
		
		System.out.println("Enter the number of rows :");
		int iRow=sobj.nextInt();
		
		System.out.println("Enter the number of columns :");
		int iCol=sobj.nextInt();
		
		int matrix[][]=new int[iRow][iCol];
		
		System.out.println("Enter the elements of matrix:");
		
		for(int i=0;i<iRow;i++)
		{
			for(int j=0;j<iCol;j++)
			{
				matrix[i][j]=sobj.nextInt();
			}
		}
		
		System.out.println("Matrix is :");
		
		for(int i=0;i<iRow;i++)
		{
			for(int j=0;j<iCol;j++)
			{
				System.out.print(matrix[i][j]+"\t");
			}
			System.out.println();
		}
		
		Matrix obj=new Matrix();
		obj.columnExchange(matrix);
	}
}